src/sortStudents.ts

```diff
@@ -1,16 +1,60 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function Average(student: Student): number {
+  const sum: number = student.grades.reduce((acc, elem) => acc + elem, 0);
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  return sum / student.grades.length;
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  return [...students].sort((a: Student, b: Student) => {
+    let studentA: Student = a;
+    let studentB: Student = b;
+
+    if (order === 'desc') {
+      studentA = b;
+      studentB = a;
+    }
+
+    switch (sortBy) {
+      case SortType.Name:
+        return studentA.name.localeCompare(studentB.name);
+
+      case SortType.Surname:
+        return studentA.surname.localeCompare(studentB.surname);
```

Comment:
You do the same in both cases. Think about combining them
```suggestion
      case SortType.Name:
      case SortType.Surname:
        return ....;
```

Reply:
+